spring-context-indexer/src/main/java/org/springframework/context/index/processor/MetadataStore.java

```diff
@@ -62,12 +62,9 @@ public void writeMetadata(CandidateComponentsMetadata metadata) throws IOExcepti
 
 
 	private CandidateComponentsMetadata readMetadata(InputStream in) throws IOException {
-		try {
+		try (in) {
```

Comment:
At a quick glance, this change is the only change in this PR that does not break existing behavior.

Can you please revisit your changes and ensure that existing behavior is not altered?

For example, a try-with-resources block will close the `AutoCloseable` object, but it will not swallow exceptions thrown by the invocation of `close()`. So anywhere that we were previously intentionally swallowing exceptions such as `IOException`, we would still have to swallow those exceptions.

In addition, we would have to be able to disambiguate between an `IOException` thrown from with the try-block vs. one thrown by the invocation of `close()`.

In light of that, I don't think it makes sense to use try-with-resources in such use cases, but I'm happy to be proven wrong.

Reply:
You are right, I mistakenly thought that exceptions in a try-with-resource were going to be swallowed. They aren't and still bubble up. Lets remove and forget about this pull request :).

Reply:
Actually, I'd appreciate it if you could keep the one valid change and add inline documentation to the other locations to point out why try-with-resources should not be used there.

That will provide a big help to the team to make sure we don't accidentally switch to try-with-resources for those use cases in the future.

Reply:
>Actually, I'd appreciate it if you could keep the one valid change and add inline documentation to the other locations to point out why try-with-resources should not be used there.

Of course if you'd prefer to simply close this PR, that's also fine.

I'll leave it up to you.

Reply:
Done. Left the one in place, placed a remark on the others.

Reply:
Thanks for making the changes. That looks good now.